pkg/auth/dependency/sso/azureadv2.go

```diff
@@ -206,11 +206,10 @@ func (f *Azureadv2Impl) OpenIDConnectGetAuthInfo(r OAuthAuthorizationResponse) (
 		err = fmt.Errorf("cannot find oid")
 		return
 	}
-	email, ok := mapClaims["email"].(string)
-	if !ok {
-		err = fmt.Errorf("cannot find email")
-		return
-	}
+	// For "Microsoft Account", email usually exists.
+	// For "AD guest user", email usually exists because to invite an user, the inviter must provide email.
+	// For "AD user", email never exists even one is provided in "Authentication Methods".
+	email, _ := mapClaims["email"].(string)
```

Comment:
If I understand correctly, most use cases for AD would fall into the "AD user" case, which email could not be obtained, so auto-linking would not work in this case.

Reply:
Yes. But we can do nothing about it.